eth2near/utilities/src/eth_proof_generator.rs

```diff
@@ -0,0 +1,243 @@
+        .append(&header.state_root.0.as_slice())
+        .append(&header.transactions_root.0.as_slice())
+        .append(&header.receipts_root.0.as_slice())
+        .append(&header.logs_bloom.0.as_slice())
+        .append(&header.difficulty)
+        .append(&header.number.0)
+        .append(&header.gas_limit.0)
+        .append(&header.gas_used.0)
+        .append(&header.timestamp.0)
+        .append(&header.extra_data.0.as_slice())
+        .append(&header.mix_hash.0.as_slice())
+        .append(&header.nonce.0.as_slice());
+
+    header.base_fee_per_gas.map(|v| stream.append(&v));
+    header.withdrawals_root
+        .as_ref()
+        .map(|v| stream.append(&v.0.as_slice()));
+    header.blob_gas_used.map(|v| stream.append(&v));
+    header.excess_blob_gas.map(|v| stream.append(&v));
+    header.parent_beacon_block_root
+        .as_ref()
+        .map(|v| stream.append(&v.0.as_slice()));
+
+    stream.finalize_unbounded_list();
+    stream.out().to_vec()
+}
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+    use hasher::Hasher;
+    const RPC_URL: &'static str = "https://eth.llamarpc.com";
+
+    #[test]
+    fn generate_proof_pre_shapella() {
+        let tx_hash = "0xc4a6c5cde1d243b26b013f805f71f6de91536f66c993abfee746f373203b68cc";
+        let proof = get_proof_for_event(tx_hash, 251, RPC_URL).unwrap();
+
+        let expected_log_index = 0;
+        let expected_receipt_index = 167;
+        let expected_header = "f61633d948772c98982af9df989c5cb9b9a8f5ed1a8411d57feb794013941de5";
+        let expected_receipt = "02f902c70183f2d2d4b9010000000000000000000000000000000000000000000000000000000000000000000800010000000000000002000100000000000000000000000000000000000000000200000000000008000008000000000000000000000000000000004000000000000000000000000000000000000000000000000000010000000010000000000000000000000000000000000000000200000000010000000000000000000000000000000000200080000000200000000000000000000000000000004000000000000002000000000000000000000000000000000000080000000000000000000000000000000000000000000200000004000000000000000000000000000000f901bcf89b94a0b86991c6218b36c1d19d4a2e9eb0ce3606eb48f863a0ddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3efa0000000000000000000000000ef013a60f765f34b0fd7c5aaf83b9c65bb10a9afa000000000000000000000000023ddd3e3692d1861ed57ede224608875809e127fa0000000000000000000000000000000000000000000000000000000174876e800f9011c9423ddd3e3692d1861ed57ede224608875809e127ff863a0dd85dc56b5b4da387bf69c28ec19b1d66e793e0d51b567882fa31dc50bbd32c5a0000000000000000000000000a0b86991c6218b36c1d19d4a2e9eb0ce3606eb48a0000000000000000000000000ef013a60f765f34b0fd7c5aaf83b9c65bb10a9afb8a0000000000000000000000000000000000000000000000000000000174876e8000000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000002f6175726f72613a343332456231663237333036363241443141393739314564333443423244424466373030313535350000000000000000000000000000000000";
```

Comment:
I'd propose to store these values in file(s) and load them during testing?

Reply:
Thanks, I agree it could be nice. I'm not sure it's worth it, however. There aren't that many of these values and keeping them close to test logic is convenient. Plus, we may want to merge this PR sooner rather than later

Reply:
Also, it will be good to describe how you get this data in the comments. 

I expect, that one day these tests will become out of date, and instruction on how to update the data for tests will save a lot of time in the future. Also, it is another reason to separate test data. So, we can change it without changing the tests itself.

Reply:
@kisialiou @olga24912  done in 518d2dae6d00756fdf8885451e61db896aeb802b